pygin/pygin/src/helpers.cpp

```diff
@@ -0,0 +1,52 @@
+#include "headers.hpp"
+
+#include "helpers.hpp"
+#include "python.hpp"
+#include "py.common.hpp"
+
+namespace helpers
+{
+
+namespace {
+struct py_deleter
+{
+	void operator ()(void* p) const { PyMem_Free(p); }
+};
+} // anonymous
+
+void py_string_storage::clear()
+{
+	while (!this->strings.empty())
+	{
+		py_deleter()(this->strings.back());
+		this->strings.pop_back();
+	}
+}
+
+wchar_t const* py_string_storage::push_back(py::string const& s)
+{
+	Py_ssize_t Size;
+	std::unique_ptr<wchar_t [], py_deleter> Data(
+		py::invoke(PyUnicode_AsWideCharString, s.get(), &Size));
+	this->strings.push_back(Data.get());
+	return Data.release();
+}
+
+wchar_t const* py_string_storage::push_back_allow_none(py::object const& o)
```

Comment:
Why is it needed?

Reply:
Because py::string cannot hold `None`. So this is for translation python str/None to whar_t*/nullptr.

Reply:
Regarding `push_back_if_not_none` - I would really want to leave it as is. Currently, it is called 5 times and besides checking for `None` it also does `cast<py::string>`. Without it, the calling side would be much more verbose. This is just a free helper function intentionally left out of `py_string_storage` class interface.